components/meter/meter-linear.js

```diff
@@ -32,23 +49,61 @@ class MeterLinear extends LitElement  {
 				position: absolute;
 				top: 0;
 			}
+
+			.d2l-meter-linear-text {
+				color: var(--d2l-color-ferrite);
+				display: flex;
+				flex-direction: row;
+				justify-content:space-between;
+				margin: 0 6px;
+			}
+
+			.d2l-meter-linear-secondary {
+				align-self: flex-end;
+			}
+
 		`];
 	}
 
 	constructor() {
 		super();
-		this.value = 0;
 		this.max = 100;
+		this.percent = false;
+		this.singleLine = false;
+		this.value = 0;
 	}
 
 	render() {
 		const progressBarWidth = this.value / this.max * 100;
+		const primary = this.percent ? `${Math.floor(this.value / this.max * 100)}%` : `${this.value}/${this.max}`;
+		const secondaryText = this._formatContext(this.context, this.value, this.max);
+		const secondaryTextElement = secondaryText ? html`<div class="d2l-meter-linear-secondary">${secondaryText}</div>` : html``;
+		const classSingleLine = this.singleLine ? 'd2l-meter-linear-flex' : 'd2l-meter-linear-block';
 		return html `
-			<div class="d2l-meter-linear-full-bar">
-				<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+			<div class="${classSingleLine}" role="figure" aria-label="${secondaryText}: ${primary}; progress indicator">
+				<div class="d2l-meter-linear-full-bar" role="presentation">
+					<div class="d2l-meter-linear-inner-bar" style="width:${progressBarWidth}%;"></div>
+				</div>
+				<div class="d2l-body-small d2l-meter-linear-text" role="presentation">
+					<!-- I am inserting this space here for visual as well copy and pasting primary and secondary text gives the correct results -->
+					<div class="d2l-meter-linear-primary" role="presentation">${primary}&nbsp;</div>
+					${secondaryTextElement}
+				</div>
 			</div>
 		`;
 	}
+
+	_formatContext(context, value, max) {
+		if (!context) {
+			return '';
+		}
+
+		// Floor so you don't get 100% when you are at 99.5%
+		context = context.replace('{%}', `${Math.floor(value / max * 100)}%`);
+		context = context.replace('{x/y}', `${value}/${max}`);
```

Comment:
This allows the user to use localization for the context text and still be able to add the data.

Reply:
Perhaps the token should be `{value/max}`?

Reply:
FYI, I don't feel strongly about this.  Initially (before I saw the replacement code) I thought this might do something else magical.  Alternatively it could be `{fraction}` or `{frac}`... leave it up to you.  :)

Reply:
I do like `{frac}` but I think I'll change it back to `{x/y}` to match design documentation. Also, I think more people will like `{x/y}` over `{frac}`? To me `{frac}` makes sense because of my years using Latex.